packages/gensx/src/component.ts

```diff
@@ -64,16 +64,74 @@ export function StreamComponent<P>(
   fn: (props: P) => MaybePromise<Streamable | JSX.Element>,
 ): GsxStreamComponent<P> {
   const GsxStreamComponent: GsxStreamComponent<P> = async props => {
-    const iterator: Streamable = await resolveDeep(fn(props));
-    if (props.stream) {
-      return iterator;
-    }
+    const context = getCurrentContext();
+    const workflowContext = context.getWorkflowContext();
+    const { checkpointManager } = workflowContext;
+
+    // Create checkpoint node for this component execution
+    const nodeId = await checkpointManager.addNode(
+      {
+        componentName: name,
+        props: Object.fromEntries(
+          Object.entries(props).filter(([key]) => key !== "children"),
+        ),
+      },
+      context.getCurrentNodeId(),
+    );
+
+    try {
+      const iterator: Streamable = await context.withCurrentNode(nodeId, () =>
+        resolveDeep(fn(props)),
+      );
+
+      if (props.stream) {
+        // Mark as streaming immediately
+        checkpointManager.completeNode(nodeId, "[streaming in progress]");
 
-    let result = "";
-    for await (const token of iterator) {
-      result += token;
+        // Create a wrapper iterator that captures the output while streaming
+        const wrappedIterator = async function* () {
+          let accumulated = "";
+          try {
+            for await (const token of iterator) {
+              accumulated += token;
+              yield token;
```

Comment:
We don't want to send updates for the intermediate values?

Reply:
Sounds good. I guess it would just add to that feeling of fast and reactive UX, to be able to watch it stream in while you are watching the workflow execute. I know I would have liked this in Langsmith for some of the long running streams we had.

Given the way we simply skip until the current update is done syncing, it will be extra writes, but not one for every token, or anything like that. But it might suggest moving to a debounced update model or something.

Happy to punt for now, but seems like a feature the reenforces that reactive experience we are going for. Maybe @danfhernandez has an opinion about how important it is.